### DISCUSSION TITLE
CTT Browse of Invalid Node receives Status Good

### DISCUSSION THREAD

**Discussion by johannwesely** [eclipse-milo/milo#1308] [General]
Hi i have the following CTT Issues i am running on  milo/Dev1.0. 
![image](https://github.com/user-attachments/assets/db527eaf-6201-4eb2-84d1-238d49bfaa89)
i also tested with milo client and have also quality good for invalid node id?

```
BrowseDescription browse = new BrowseDescription(new NodeId(1, "/Humbug"), BrowseDirection.Forward, NodeIds.References, true, uint(NodeClass.Object.getValue() | NodeClass.Variable.getValue()), uint(BrowseResultMask.All.getValue()));

BrowseResult browseResult = client.browse(browse);

```
Any Suggestions ?
Greetings Johann

**Comment by kevinherron** [comment]
Looks like I broke this in either 934aa0b58c0ca322155db80762c98292fa09ac57 or a79fe4613e346db08c2f3bef96f34c6b00e10420. I'll try to figure out a fix soon.

**Reply by kevinherron** [reply]
The distinction between a `ReferenceResult.BadNodeIdUnknown` and no references seems to be lost here: https://github.com/eclipse/milo/blob/318bf4e8a0b303e482571ed807afc27bce056502/opc-ua-sdk/sdk-server/src/main/java/org/eclipse/milo/opcua/sdk/server/servicesets/impl/helpers/BrowseHelper.java#L92-L104

**Comment by kevinherron** [comment]
Ok, this should be fixed now...

I'm seeing some other failures in the View Basic set but can't figure out if it's a CTT configuration problem or something else. All the failures have a strange `Ctt_Bad` status listed in the failure. It seems isolated to the ones that need "Has References of a ReferenceType and SubType" configured.

<img width="948" alt="image" src="https://github.com/user-attachments/assets/b8e4036c-08ee-48d6-a5da-51587caf03df">

**Reply by johannwesely** [reply]
Hi Kevin,
thank you i test this today. the 020 case is skipped in my test because i didnt config a "Has References of a ReferenceType and Subtype" in Settings. I Dont know if this is crucial or not ? But i have more issues in View Basic should i start a new discussion for them?

**Reply by johannwesely** [reply]
fyi: the fix worked.
